src/GAP/iso_oscar_gap.jl

```diff
@@ -55,12 +55,18 @@ function _iso_oscar_gap(RO::Union{Nemo.zzModRing, Nemo.ZZModRing})
    return MapFromFunc(RO, RG, f, finv)
 end
 
-# Assume that `FO` and `FG` are finite fields of the same order
-# in Oscar and GAP, respectively.
-function _iso_oscar_gap_field_finite_functions(FO::Union{Nemo.fpField, Nemo.FpField}, FG::GAP.GapObj)
+_ffe_to_int(a::FqFieldElem) = Nemo._coeff(a, 0)
+_ffe_to_int(a::FqPolyRepFieldElem) = coeff(a, 0)
+_ffe_to_int(a::fqPolyRepFieldElem) = ZZ(coeff(a, 0))
```

Comment:
@ThomasBreuer I assume you added the conversion to `ZZRingElem` because we otherwise get a `UInt`.

We should have a better function to "convert a Julia or Oscar integer to GAP integer" which does the right thing for UInt (i.e. convert it to an `Int` it if fits and otherwise to a GAP larger integer). A natural way to express that would be via `GapInt(x)` methods

Reply:
Yes, the other `coeff` and `_coeff` methods return  `ZZRingElem`, and here we get a `UInt`. This is not really a problem if we are sure that anyhow `GAP.Obj` will be called for the result. Thus perhaps we do not need `_ffe_to_int` but `_ffe_to_gapint`?

Reply:
Yeah, and we can use `julia_to_gap` to get "efficient" conversion for UInt, too (as long as they are less than $2^{60}$ at least; for larger `UInt` it still first converts to a BigInt. I could optimize that, too, but that's for GAP.jl